Framework/src/InfrastructureSpecReader.cxx

```diff
@@ -91,6 +92,7 @@ TaskSpec InfrastructureSpecReader::readSpecEntry<TaskSpec>(const std::string& ta
   TaskSpec ts;
 
   ts.taskName = taskTree.get<std::string>("taskName", taskID);
+  validateName(ts.taskName); // bail if the 16 first characters of the name match another device
```

Comment:
But you are using `taskName`, `checkName`, etc instead of deviceName, but a check called `Tracks` will not be in a device named the same as a QC task (e.g. `qc-check-TST-Tracks` vs `qc-task-TST-Tracks`).

You are also preventing us from having tasks with the same name from two different detectors.

What would make more sense is to check if there are any duplicate output specs in the workflow.